python/paddle/fluid/tests/unittests/test_dropout_op.py

```diff
@@ -82,5 +83,37 @@ def test_check_output(self):
         self.check_output()
 
 
+class TestFP16DropoutOp1(OpTest):
+    def setUp(self):
+        x = np.random.random((32, 64)).astype("float16")
+        prob = 0.35
+        out = x * (1.0 - prob)
+
+        self.op_type = "dropout"
+        self.inputs = {'X': OpTest.np_dtype_to_fluid_dtype(x)}
+        self.attrs = {'dropout_prob': prob, 'fix_seed': True, 'is_test': True}
+        self.outputs = {'Out': out}
+
+    def test_check_output(self):
+        if core.is_compiled_with_cuda() and core.op_support_gpu("dropout"):
+            self.check_output_with_place(core.CUDAPlace(0), atol=1e-3)
+
+
+class TestFP16DropoutOp2(OpTest):
+    def setUp(self):
+        x = np.random.random((32, 64, 3)).astype("float16")
+        prob = 0.75
+        out = x * (1.0 - prob)
+
+        self.op_type = "dropout"
+        self.inputs = {'X': OpTest.np_dtype_to_fluid_dtype(x)}
+        self.attrs = {'dropout_prob': prob, 'is_test': True}
+        self.outputs = {'Out': out}
+
+    def test_check_output(self):
+        if core.is_compiled_with_cuda() and core.op_support_gpu("dropout"):
+            self.check_output_with_place(core.CUDAPlace(0), atol=1e-3)
```

Comment:
`TestFP16DropoutOp1` and `TestFP16DropoutOp2` are very similar, and they can be inherited relationships, which can reduce the amount of code.

Reply:
You are right! Done.